app/screens/create_channel/create_channel.js

```diff
@@ -154,7 +155,7 @@ export default class CreateChannel extends PureComponent {
         if (event.type === 'NavBarButtonPress') {
             switch (event.id) {
             case 'close-new-channel':
-                this.close(this.props.channelType === General.OPEN_CHANNEL);
+                this.close(this.props.channelType === General.OPEN_CHANNEL && !this.props.isModal);
```

Comment:
Can we simplify this to just `!this.props.isModal`? With the new `isModal` flag, I don't think we're tied to the `channelType` explicitly anymore.

Reply:
Just refactored this and we can actually get away with only using the passed in `closeButton` prop to conditionally render the button. `isModal` can be removed completely.